### ISSUE TITLE
Weird behaviour with configureRoutes, childAdmin and dev-cache ?

### ISSUE THREAD

**Issue by VincentLanglet** [sonata-project/SonataAdminBundle#5946]
#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
symfony/asset                    v4.4.4  v5.0.4  Symfony Asset Component
symfony/browser-kit              v4.4.4  v5.0.4  Symfony BrowserKit Component
symfony/cache                    v4.4.4  v5.0.4  Symfony Cache component with PSR-6, PSR-16, and tags
symfony/config                   v4.4.4  v5.0.4  Symfony Config Component
symfony/console                  v4.4.4  v5.0.4  Symfony Console Component
symfony/contracts                v1.1.8  v2.0.1  A set of abstractions extracted out of the Symfony components
symfony/css-selector             v4.4.4  v5.0.4  Symfony CssSelector Component
symfony/debug                    v4.4.4  v4.4.4  Symfony Debug Component
symfony/debug-bundle             v4.4.4  v5.0.4  Symfony DebugBundle
symfony/dependency-injection     v4.4.4  v5.0.4  Symfony DependencyInjection Component
symfony/doctrine-bridge          v4.4.4  v5.0.4  Symfony Doctrine Bridge
symfony/dom-crawler              v5.0.4  v5.0.4  Symfony DomCrawler Component
symfony/dotenv                   v4.4.4  v5.0.4  Registers environment variables from a .env file
symfony/error-handler            v4.4.4  v5.0.4  Symfony ErrorHandler Component
symfony/event-dispatcher         v4.4.4  v5.0.4  Symfony EventDispatcher Component
symfony/expression-language      v4.4.4  v5.0.4  Symfony ExpressionLanguage Component
symfony/filesystem               v4.4.4  v5.0.4  Symfony Filesystem Component
symfony/finder                   v4.4.4  v5.0.4  Symfony Finder Component
symfony/flex                     v1.6.2  v1.6.2  Composer plugin for Symfony
symfony/form                     v4.4.4  v5.0.4  Symfony Form Component
symfony/framework-bundle         v4.4.4  v5.0.4  Symfony FrameworkBundle
symfony/http-foundation          v4.4.4  v5.0.4  Symfony HttpFoundation Component
symfony/http-kernel              v4.4.4  v5.0.4  Symfony HttpKernel Component
symfony/inflector                v5.0.4  v5.0.4  Symfony Inflector Component
symfony/intl                     v4.4.4  v5.0.4  A PHP replacement layer for the C intl extension that includes additional data from the ICU library.
symfony/mime                     v5.0.4  v5.0.4  A library to manipulate MIME messages
symfony/monolog-bridge           v5.0.4  v5.0.4  Symfony Monolog Bridge
symfony/monolog-bundle           v3.5.0  v3.5.0  Symfony MonologBundle
symfony/options-resolver         v4.4.4  v5.0.4  Symfony OptionsResolver Component
symfony/phpunit-bridge           v4.4.4  v5.0.4  Symfony PHPUnit Bridge
symfony/polyfill-ctype           v1.14.0 v1.14.0 Symfony polyfill for ctype functions
symfony/polyfill-iconv           v1.14.0 v1.14.0 Symfony polyfill for the Iconv extension
symfony/polyfill-intl-grapheme   v1.14.0 v1.14.0 Symfony polyfill for intl's grapheme_* functions
symfony/polyfill-intl-icu        v1.14.0 v1.14.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn        v1.14.0 v1.14.0 Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-intl-normalizer v1.14.0 v1.14.0 Symfony polyfill for intl's Normalizer class and related functions
symfony/polyfill-mbstring        v1.14.0 v1.14.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72           v1.14.0 v1.14.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-php73           v1.14.0 v1.14.0 Symfony polyfill backporting some PHP 7.3+ features to lower PHP versions
symfony/polyfill-uuid            v1.14.0 v1.14.0 Symfony polyfill for uuid functions
symfony/process                  v4.4.4  v5.0.4  Symfony Process Component
symfony/property-access          v4.4.4  v5.0.4  Symfony PropertyAccess Component
symfony/routing                  v4.4.4  v5.0.4  Symfony Routing Component
symfony/security-acl             v3.0.4  v3.0.4  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle          v4.4.4  v5.0.4  Symfony SecurityBundle
symfony/security-core            v4.4.4  v5.0.4  Symfony Security Component - Core Library
symfony/security-csrf            v4.4.4  v5.0.4  Symfony Security Component - CSRF Library
symfony/security-guard           v4.4.4  v5.0.4  Symfony Security Component - Guard
symfony/security-http            v4.4.4  v5.0.4  Symfony Security Component - HTTP Integration
symfony/serializer               v4.4.4  v5.0.4  Symfony Serializer Component
symfony/stopwatch                v4.4.4  v5.0.4  Symfony Stopwatch Component
symfony/string                   v5.0.4  v5.0.4  Symfony String component
symfony/swiftmailer-bundle       v3.4.0  v3.4.0  Symfony SwiftmailerBundle
symfony/templating               v4.4.4  v5.0.4  Symfony Templating Component
symfony/translation              v4.4.4  v5.0.4  Symfony Translation Component
symfony/twig-bridge              v4.4.4  v5.0.4  Symfony Twig Bridge
symfony/twig-bundle              v4.4.4  v5.0.4  Symfony TwigBundle
symfony/validator                v4.4.4  v5.0.4  Symfony Validator Component
symfony/var-dumper               v4.4.4  v5.0.4  Symfony mechanism for exploring and dumping PHP variables
symfony/var-exporter             v5.0.4  v5.0.4  A blend of var_export() + serialize() to turn any serializable data structure to plain PHP code
symfony/web-profiler-bundle      v4.4.4  v5.0.4  Symfony WebProfilerBundle
symfony/yaml                     v4.4.4  v5.0.4  Symfony Yaml Component
```

#### Symfony packages

```
$ composer show --latest 'symfony/*'
sonata-project/admin-bundle              3.61.0 3.61.0 The missing Symfony Admin Generator
sonata-project/block-bundle              3.18.4 4.1.0  Symfony SonataBlockBundle
sonata-project/cache                     2.0.1  2.0.1  Cache library
sonata-project/core-bundle               3.18.0 3.18.0 Symfony SonataCoreBundle (abandoned)
sonata-project/datagrid-bundle           2.5.0  3.1.1  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.5.1  1.5.1  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.14.0 3.14.0 Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/exporter                  2.1.0  2.1.0  Lightweight Exporter library
```

#### PHP version

```
$ php -v
PHP 7.4.2 (cli) (built: Feb  2 2020 07:56:32) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.2, Copyright (c), by Zend Technologies
```

## Subject

One developper in our worked with childAdmin.
Let's say we have a SubscriptionAdmin, with a child IncidentAdmin.
Both extends an AbstractAdmin with the following code: 
```
/**
 * @param RouteCollection $collection
 */
protected function configureRoutes(RouteCollection $collection)
{
    if ($this->isChild()) {
        $collection->clearExcept(['create']);
    } else {
        $collection
            ->remove('create')
            ->remove('export')
            ->remove('delete');
    }
}
```

My first problem here is the following, when trying to create a child IncidentAdmin I get an error because subscription_incident_list route does not exist. The route is necessary because of the clickable breadcrum: `/subscriptionList/subscriptionId/incidentList/createAnIncident`

There should be a check in the breadcrum, and display only something like `/subscriptionList/subscriptionId/createAnIncident` if the list does not exist.

So I temporally fixed this with
```
/**
 * @param RouteCollection $collection
 */
protected function configureRoutes(RouteCollection $collection)
{
    if ($this->isChild()) {
        $collection->clearExcept(['create', 'list']);
    } else {
        $collection
            ->remove('create')
            ->remove('export')
            ->remove('delete');
    }
}
```

But I had a really really weird behaviour. The bug was still here !
Then I've overriden the configureRoutes directly in the IncidentAdmin. And the bug was gone.
Then I remove the override the configureRoutes in the IncidentAdmin. And it still worked.

And if I changed `$collection->clearExcept(['create', 'list']);` by `$collection->clearExcept(['create']);`, it was still working. 
But when I modified the IncidentAdmin configureRoutes, refresh, then removed the code added, and refresh again, it stopped working.

It was like there was some cache in the route configuration... Is it possible it come from Sonata ?

**Comment by VincentLanglet** [comment]
And now, when we clicked on the List breadcrumb, we get an error because the `subscription_incident_batch` route does not exist.

So, In order to allow `create` I have to allow `list`, but in order to allow `list` I have to allow `batch`.

And then if I created one Incident, in order to display the `list`, I have to allow `show`.
I even get an error because `export` is missing. But I don't want it !

I think a lot of checks are missing.
